internal/rbd/rbd_journal.go

```diff
@@ -233,8 +233,8 @@ func (rv *rbdVolume) Exists(ctx context.Context, parentVol *rbdVolume) (bool, er
 	}
 
 	kmsID := ""
-	if rv.Encrypted {
-		kmsID = rv.KMS.GetID()
+	if rv.isEncrypted() {
```

Comment:
incase if csidriver gets restarted between the rbd image creation and metadata store. are we handling that scenario also? as we are checking the image exists we also need to make sure the metadata is also set.

Reply:
This behaviour is not changed. The metadata of the image already contains the KMS-ID, it would be problematic if that is missing. We are adding more things in the metadata with the SecretsMetadataKMS, so the potential issue more likely to cause problems when that KMS is in use.

I think #1881 will be used for enhancing logging, and maybe tracking a status/phase of a volume while provisioning. It would be possible to detect incomplete provisioned volumes that way. IMHO it is not something this PR can address.